### PR TITLE
Add package abstraction

### PR DESCRIPTION
**Pull request by bregman-arie**
This will allow us to easily access package attributes
in addition to running some operations seemingly like
printing out information on the package.

### CODE REVIEW

**Review by sdatko — CHANGES_REQUESTED** [review]
This breaks now the generator... IMO introducing the Package data class package should not require that many changes in browser.

Lets jump on google meet to address this.

**Comment by bregman-arie** [comment]
Yes, as I told you, it's initial work, don't expect it to work at the moment :)
Just looking for initial feedback before proceeding to generator

File: znoyder/package.py [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+            packages = [package for package in packages
+                        if filters.get('tag') in package.tags]
+        if filters.get('upstream'):
+            packages = [package for package in packages
+                        if filters.get('upstream') in str(package.upstream)]
+        return packages
+
+    def __str__(self):
+        return "{} {} {}".format(self.osp_name, self.osp_distgit,
+                                 self.osp_patches)
```
Comment (sdatko) [inline znoyder/package.py:52]: Just osp_name or osp_project here

Comment (bregman-arie) [reply znoyder/package.py]: Done

**Review by IgnacioPro — APPROVED** [review]
Looks good to me.

**Comment by bregman-arie** [comment]
Closing this one since there was not enough support for merging it and there was a decision to focus on other aspects of the project like writing unit tests for the existing codebase.

**Pull request closed without merging**